test/unittest/rtps/history/BasicPoolsTests.cpp

```diff
@@ -34,7 +34,7 @@ class BasicPoolsTest : public TestWithParam<tuple<uint32_t, uint32_t, uint32_t,
         : pool_size_(10)
         , max_pool_size_(0)
         , payload_size_(128)
-        , memory_policy_(MemoryManagementPolicy::PREALLOCATED_MEMORY_MODE)
+        , memory_policy_(MemoryManagementPolicy::PREALLOCATED_WITH_REALLOC_MEMORY_MODE)
```

Comment:
Again, this is not checking the default behavior. It is an initialization but every case is going to be tested. Thus, I would suggest not modifying it.